### ISSUE TITLE
Issue #2 – Protein Dataset Pipeline

### ISSUE THREAD

**Issue by mattackerman1** [mattackerman1/mini-alphafold#2]
- Research & choose initial datasets (secondary structure, small PDB subset)
- Implement ProteinDataset class (PyTorch)
- Support sequence tokenization (amino acids → integers)
- Add DataLoader with padding & batching
